lib/notifier/index.js

```diff
@@ -166,6 +166,50 @@ function transactionNotify (tx, rec) {
   })
 }
 
+function complianceNotify (customer, deviceId, ...args) {
+  return Promise.all([
+    settingsLoader.loadLatest(),
+    queries.getMachineName(deviceId)
+  ])
+    .then(([settings, machineName]) => {
+      const notifications = configManager.getGlobalNotifications(settings.config)
+
+      const [action, period] = args
+
+      const msgCore = {
+        BLOCKED: `was blocked`,
+        SUSPENDED: `was suspended for ${!!period && period} days`
+      }
+
+      const rec = {
+        sms: {
+          body: `Customer ${customer.phone} ${msgCore[action]} - ${machineName}`
+        },
+        email: {
+          subject: `Customer compliance`,
+          body: `Customer ${customer.phone} ${msgCore[action]} in machine ${machineName}`
+        }
+      }
+
+      const promises = []
+
+      const emailActive =
+        notifications.email.active &&
+        notifications.email.compliance
+
+      const smsActive =
+        notifications.sms.active &&
+        notifications.sms.compliance
+
+      if (emailActive) promises.push(emailFuncs.sendMessage(settings, rec))
+      if (smsActive) promises.push(smsFuncs.sendMessage(settings, rec))
+
+      notifyIfActive('compliance', 'customerComplianceNotify', customer, deviceId, ...args)
```

Comment:
Also it's better to just pass what's needed here, this rest param stuff can get complicated pretty fast